plugin/builtin/replicacounttransformer/ReplicaCountTransformer.go

```diff
@@ -31,30 +29,46 @@ func (p *plugin) Config(
 	ldr ifc.Loader, rf *resmap.Factory, c []byte) (err error) {
 
 	p.Replica = types.Replica{}
+	p.FieldSpecs = nil
 	return yaml.Unmarshal(c, p)
 }
 
 func (p *plugin) Transform(m resmap.ResMap) error {
-	matcher := func(r resid.ResId) bool {
-		return r.ItemId.Name == p.Replica.Name
+	for i, replicaSpec := range p.FieldSpecs {
+		for _, id := range m.GetMatchingIds(p.createMatcher(i)) {
+			err := transformers.MutateField(
+				m.GetById(id).Map(), replicaSpec.PathSlice(),
+				replicaSpec.CreateIfNotPresent, p.addReplicas)
+			if err != nil {
+				return err
+			}
+		}
 	}
 
-	for _, id := range m.GetMatchingIds(matcher) {
-		kMap := m.GetById(id).Map()
+	return nil
+}
 
-		specInterface, ok := kMap[fldSpec]
-		if !ok {
-			return fmt.Errorf("object %s missing field %s, cannot update %s",
-				p.Replica.Name, fldSpec, fldReplica)
-		}
+// Match Replica.Name and FieldSpec
+func (p *plugin) createMatcher(i int) resmap.IdMatcher {
+	return func(r resid.ResId) bool {
+		return r.ItemId.Name == p.Replica.Name &&
+			r.Gvk().IsSelected(&p.FieldSpecs[i].Gvk)
+	}
+}
 
-		if spec, ok := specInterface.(map[string]interface{}); ok {
-			spec[fldReplica] = p.Replica.Count
-			kMap[fldSpec] = spec
-		} else {
-			return fmt.Errorf("object %s has a malformed %s", p.Replica.Name, fldSpec)
+func (p *plugin) addReplicas(in interface{}) (interface{}, error) {
+	switch m := in.(type) {
+	case int64:
+		// Was already in the field.
+	case map[string]interface{}:
```

Comment:
I don't really like the way this is implemented, but wasn't sure if there was a better way.